### PR TITLE
Support FQN in belongsToMany through option

### PR DESCRIPTION
**Pull request by andrii-pukhalevych**
Support FQN in `belongsToMany` `through` option:

`$this->belongsToMany('CustomerPersons', ['className' => PersonsTable::class, 'through' => CompaniesCustomersTable::class]);`

CakePhp supports FQNs here, but IDE Helper generates wrong property name in the annotation

### CODE REVIEW

**Comment by andrii-pukhalevych** [comment]
Code to get name for FQN comes from here:

https://github.com/cakephp/cakephp/blob/a3e4dec82024bb4bf5e17d804c27cc44b69fa931/src/ORM/Locator/TableLocator.php#L219

File: src/Annotator/ModelAnnotator.php [hunk 1]
```diff
@@ -280,6 +280,10 @@ protected function getAssociations(AssociationCollection $tableAssociations): ar
 
 			$className = App::className($through, 'Model/Table', 'Table') ?: static::CLASS_TABLE;
 			[, $throughName] = pluginSplit($through);
+			if (strpos($throughName, '\\') !== false) {
+				/** @psalm-suppress PossiblyFalseOperand */
```
Comment (dereuromark) [inline src/Annotator/ModelAnnotator.php:284]: I dont think you need this for phpstan

**Comment by dereuromark** [comment]
I wonder if we could add a test case for this specific issue.

**Comment by andrii-pukhalevych** [comment]
Please add test for this. I tried to do but its bit unclear for me how to add it for this case

**Pull request merged**